### PR TITLE
blog: mapping location on dashboard 2.0

### LINKED ISSUE #2015: Blog: Mapping Locations on Dashboard 2.0
### Description

Write a blog post detailing how to obtain location data (latitude, longitude) and plot it on a map within Dashboard 2.0.

### LINKED ISSUE #2031: Blog Art : mapping location on Dashboard 2.0
### Related Issue

_No response_

### Description

**Considerations:**
- Blog tile: Mapping location data within Node-RED Dashboard 2.0.

**Content reference:**
https://github.com/FlowFuse/website/pull/2030/files

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Related Issue(s)

closes #2015 
closes #2031 

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by Yndira-E** [comment]
Hi @sumitshinde-84 here it is 😉 
mapping-location-dashboard-2

File: src/blog/2024/05/mapping-location-on-dashboard-2.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+## Formatting Location Data
+
+To ensure compatibility with the Worldmap custom node, we need to format the location data appropriately. 
+
+1. Drag the **Change** node onto the canvas, and set the `msg.payload` to `msg.payload.vehicles`.
+
+!["Screenshot of the Change node setting the payload to the vehicles JSON array containing actual vehicle location data"](./images/mapping-location-on-dashboard-2-change-node.png "Screenshot of the Change node setting the payload to the vehicles JSON array containing actual vehicle location data"){data-zoomable}
+
+2. Drag a **Function** node onto the canvas and paste the following code into it.
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/mapping-location-on-dashboard-2.md:50]: No, this is a split and join node sandwiching some other nodes. Among which a change node. Please keep things low code, youre not showing off your JS knowledge but how capable you are keeping things close to no code

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: src/blog/2024/05/mapping-location-on-dashboard-2.md (hunk 1 above)
Comment (ZJvandeWeg) [reply src/blog/2024/05/mapping-location-on-dashboard-2.md:50]: @sumitshinde-84 can you add an example to the split core node doc that explains how to handle mapping a collection?

Comment (sumitshinde-84) [reply src/blog/2024/05/mapping-location-on-dashboard-2.md:50]: Yes Zj, I will !

Comment (sumitshinde-84) [reply src/blog/2024/05/mapping-location-on-dashboard-2.md]: > @sumitshinde-84 can you add an example to the split core node doc that explains how to handle mapping a collection?

I have now updated the blog and flow to use core nodes.

**Comment by joepavitt** [comment]
Can we hold off on publishing this please? I want to try and make some improvements on the workflow/integration here, at least with an `iframe` node - had 4 people ping me just this week asking for better way to work with worldmap

**Comment by joepavitt** [comment]
Okay, @sumitshinde-84 fyi - I've just published https://flows.nodered.org/node/@flowfuse/node-red-dashboard-2-ui-iframe which will remove the need for a `ui-template` here. Still not as smooth as Dashboard 1.0, but removes the need for users to code, so think it's a big improvement

**Comment by sumitshinde-84** [comment]
Appreciate the update, @joepavitt! I'll update the blog to utilize it. Thanks for the improvement!

File: src/blog/2024/05/mapping-location-on-dashboard-2.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+authors: ["sumit-shinde"]
+image: /blog/2024/05/images/mapping-location-on-dashboard-2-worldmap.png
+tags:
+   - posts
+   - dashboard
+   - fleet tracking
+   - location
+---
+
+Have you ever needed to develop a fleet management system or something similar that requires a map with plotted location data on a dashboard? If so, this guide is for you. In this guide, we will learn how to plot location data on a world map within Dashboard 2.0.
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/mapping-location-on-dashboard-2.md:15]:
```suggestion
Fleet management in IoT uses sensors and software to collect real-time data on vehicles, such as location, fuel consumption, and driver behavior. This data allows companies to optimize routes, reduce costs, improve safety, and enhance overall operational efficiency of their fleet. Building an application that allows the tracking of location to support Fleet management is what this post is about.
```

File: src/blog/2024/05/mapping-location-on-dashboard-2.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+1. Drag an **iframe** widget onto the canvas.
+2. Select **ui-group** and **ui-base** for it, where you want to render the webpage.
+3. Set height and width for it according to your preference and set URL to **/worlmap**.
+
+!["Screenshot showing configurations of iframe widget for rendering worlmap page on dashboard"](./images/mapping-location-on-dashboard-2-iframe-widget.png "Screenshot showing configurations of iframe widget for rendering worlmap page on dashboard"){data-zoomable}
+
+## Deploying the flow
+
+!["Image displaying live locations of UK public transport on the dashboard"](./images/mapping-location-on-dashboard-2-uk-live-transport.gif "Image displaying live locations of UK public transport on the dashboard"){data-zoomable}
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/mapping-location-on-dashboard-2.md:103]: Follow up to include meta data? E.g. can you click on a bus to show some additional data?

File: src/blog/2024/05/mapping-location-on-dashboard-2.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
+---
+title: Mapping location data within Node-RED Dashboard 2.0.
+subtitle: Step-by-step guide to plot location data on dashboard 2.0.
+description: Learn how to plot location data on Dashboard 2.0 with this comprehensive step-by-step guide. 
+date: 2024-05-06
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/mapping-location-on-dashboard-2.md:5]: If you request a review, could you first review yourself?

```suggestion
date: 2024-05-13
```

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

**Review by ZJvandeWeg — APPROVED** [review]

**Pull request merged**